src/Pages/Settings/Kodi/Interface.elm

```diff
@@ -29,34 +42,573 @@ type alias Params =
+                |> Dropdown.setSelected (getSelectedOptionFromList 16 shared.settings_list)
+      , timezoneSelected = getTextFieldVal 16 shared.settings_list
+      , screensaverDropdown =
+            Dropdown.init
+                |> Dropdown.id "screensaver"
+                |> Dropdown.optionsBy .label (getOptionList 17 shared.settings_list)
+                |> Dropdown.setSelected (getSelectedOptionFromList 17 shared.settings_list)
+      , screensaverSelected = getTextFieldVal 17 shared.settings_list
+      , performDropdown =
+            Dropdown.init
+                |> Dropdown.id "perform"
+                |> Dropdown.optionsBy .label (getOptionList 25 shared.settings_list)
+                |> Dropdown.setSelected (getSelectedOptionFromList 25 shared.settings_list)
+      , performSelected = getTextFieldVal 25 shared.settings_list
+      , startupDropdown =
+            Dropdown.init
+                |> Dropdown.id "startup"
+                |> Dropdown.optionsBy .label (getOptionList 26 shared.settings_list)
+                |> Dropdown.setSelected (getSelectedOptionFromList 26 shared.settings_list)
+      , startupSelected = getTextFieldVal 26 shared.settings_list
+      , rssToggle = getBoolVal 9 shared.settings_list
+      , visualToggle = getBoolVal 21 shared.settings_list
+      , videoToggle = getBoolVal 22 shared.settings_list
+      , startupToggle = getBoolVal 24 shared.settings_list
+      , zoomTextField = getTextFieldVal 8 shared.settings_list
+      , editTextField = getTextFieldVal 10 shared.settings_list
+      , waitTextField = getTextFieldVal 20 shared.settings_list
+      }
+    , sendAction """{"jsonrpc": "2.0", "method": "Settings.GetSettings", "params": {}, "id": 1 }"""
+    )
 
 
 
 -- UPDATE
 
 
+settingsInputBlock : String -> (String -> msg) -> String -> String -> Element msg
+settingsInputBlock val msg title description =
+    column [ paddingEach { top = 0, bottom = 30, left = 20, right = 20 } ]
+        [ row [ paddingEach { top = 0, bottom = 20, left = 0, right = 0 } ]
+            [ paragraph [ width (px 300), Font.size 14, Font.color (rgb255 3 3 3), Font.medium ] [ text title ]
+            , Input.text [ width (px 400), Font.size 14, Font.color (rgb255 3 3 3), Border.widthEach { top = 0, bottom = 1, left = 0, right = 0 }, Border.rounded 0, Font.medium, Border.color (rgb255 3 3 3), Font.size 13 ]
+                { onChange = msg
+                , text = val
+                , placeholder = Nothing
+                , label = Input.labelHidden ""
+                }
+            ]
+        , row []
+            [ el [ width (px 300) ] (text "")
+            , paragraph [ width (px 400), Font.size 12, Font.color (rgb255 142 142 142) ] [ text description ]
+            ]
+        ]
+
+
+settingsDropdownBlock : Dropdown Option -> (Dropdown.Msg Option -> Msg) -> String -> String -> Element Msg
+settingsDropdownBlock dropdown msg title description =
```

Comment:
These functions seem to be used in multiple files and just match each other. Move them out of each file into one place, that can be referenced and used. So that we only have to maintain it once.

Reply:
This function takes `Msg` as argument and `Msg` is different for all pages so it cannot be moved to a common place.